tests/test_check_obsfiles_ert_yaml.py

```diff
@@ -245,14 +251,41 @@ def test_check_obsfiles_ert_yaml() -> None:
         yaml=True,
     )
 
-    num_dates = len(schedule.get_dates())
+    dt_schedule = pd.to_datetime(schedule.get_dates())
+
+    # Resampling dates based on requested frequency with
+    # no interpolation: keeps nearest existing date
+    if config.flownet.data_source.simulation.resampling:
+        dt_resampled = pd.date_range(
+            schedule.get_dates()[1],
+            schedule.get_dates()[-1],
+            freq=config.flownet.data_source.simulation.resampling,
+        )
+        df_schedule = pd.DataFrame(data=range(len(dt_schedule)), index=dt_schedule)
+        idx = np.zeros(len(dt_resampled), dtype=int)
+        for i, k in enumerate(dt_resampled):
+            idx[i] = df_schedule.index.get_loc(k, method="nearest")
+        dates = [
+            d.date() for d in df_schedule.iloc[np.unique(idx)].index.to_pydatetime()
+        ]
+    else:
+        dates = schedule.get_dates()
```

Comment:
The test should then actually test this code - do not re-write the exact same code here (rather make it into a function in the other file and test whether the function returns what you expect).